### ISSUE TITLE
Endless game restart (Solution found) - Dont `allowEdits(InFiles)` on mods files

### ISSUE THREAD

**Issue by Norepinephrinee** [Skidamek/AutoModpack#266]
### Bug description

Hello!

I have a dedicated server running with a large modpack. All mods are installed on client (through automodpack), still the mod asks to restart the game (the message "Successfully updated Automodpack" appears and restart is the only option). Endlessly. Nothing worked, some people have the issue, some don't. I didn't touch the config file and I tried to switch or update the mod or the forge version, still nothing helps. At first I thought there may be a duplicates of mods causing this, so I went through all of them and made sure that's not the case. The issue persists. 

Forge version is 47.3.5, I've tested it with and without performance mods, they are not the problem. The first time I've encountered such bug I've been using Forge 47.3.1.

At the present time I use the latest version of Automodpack.

I have a log, but it is kinda huge and contains a lot of errors from other mods (which I'm trying to fix).

There was also a duplicate "modpack" folder in automodpack folder which appeared in some of my attempts to fix the issue, just for the record?

### Steps to reproduce

1. Install ~120+ mods on server
2. Install Automodpack on server and client
3. Lauch the client
4. Download the mods
5. Close the game
6. Start the game
7. Restart the game after "Succesfully updated Automodpack!" message
8. Restart the game after "Succesfully updated Automodpack!" message
9. Restart the game after "Succesfully updated Automodpack!" message
...∞. Restart the game after "Succesfully updated Automodpack!" message...

### Expected behavior

Game restarts succesfully for once and then launches.

### Actual behavior

Game askes to restart every time it launches even though nothing changed.

### Relevant logs

[latest (13).log](https://github.com/user-attachments/files/16651787/latest.13.log)

### Minecraft & Mod Loader versions

1.20.1 Forge 47.3.5

### Minecraft launcher

TLauncher

### Operating system

Windows 10

### AutoModpack version

4.0.0beta14

### Other information

I have a list of mods installed if that actually helps.

Doesn't contain some farmer's delight addons, client side mods and performance mods (I've tested without them, they don't trigger the problem, something else does).
[ALL MODS.txt](https://github.com/user-attachments/files/16651739/ALL.MODS.txt)

### Check list

- [X] I have verified that the issue persists in the latest version of the mod.
- [X] I have searched the existing issues and confirmed that this is not a duplicate.

**Comment by Skidamek** [comment]
please provide server config, log from client and generated automodpack-content.json file from server

**Comment by N-Montana** [comment]
As the person who actually reported this appears to have fallen down a well, I'll do it instead.

I've attached the log from my client, the server config (which to test is identical on both client and server), the content file, and the popup that appears every time the game is launched.

Whatever this bug is stops the game from launching altogether, I've tried recopying the files out of the automodpack folder back into the main modpack folder and manually changing the client config to disable the pack, but both instances, and any other changes I make, just causes automodpack to redownload and get stuck in the same loop again.

[automodpack-content.json](https://github.com/user-attachments/files/16844776/automodpack-content.json)
[automodpack-server.json](https://github.com/user-attachments/files/16844778/autom
[debug.log](https://github.com/user-attachments/files/16844780/debug.log)
[latest.log](https://github.com/user-attachments/files/16844781/latest.log)
odpack-server.json)
![image](https://github.com/user-attachments/assets/07027d1d-1dae-4591-8c81-c93eec68fb9b)

**Comment by Skidamek** [comment]
@N-Montana all seems to be correct, but it should show that popup only once. 
Which launcher are you using? 
I would guess that launcher you are using somehow breaks it (copies the mods again as soon as automodpack deletes them - automodpack does it [on purpose](https://github.com/Skidamek/AutoModpack/wiki/File-Structure-&-Logic)) 

Could you try different one? 
I am pretty sure it works fine on modrinth, prism or official launchers.

**Comment by N-Montana** [comment]
I'm using the basic Minecraft launcher, which is why I'm confused! If I watch the folders I click play, all the mods appear, disappear, and then the popup happens.

**Comment by Skidamek** [comment]
huh, maybe thats not very professional but could i join your server and try to debug myself? you can dm me on discord [@skidam](https://discordapp.com/users/464522287618457631)

**Comment by N-Montana** [comment]
Yeah no problem... Gimme a sec I'll have to readd the mod server side

**Comment by Skidamek** [comment]
Solution: dont add `mods` to the editable files on server

**Comment by EvasiveUpgrade** [comment]
> Solution: dont add `mods` to the editable files on server

would you be able to elaborate a bit? im running a server through a website and im not really understanding what is meant by editable files

**Comment by Skidamek** [comment]
> > Solution: dont add `mods` to the editable files on server
> 
> would you be able to elaborate a bit? im running a server through a website and im not really understanding what is meant by editable files

Yes, sure take a look at the [server config](https://github.com/Skidamek/AutoModpack/wiki/Configuration#server-config-file).
And by editable files I meant the `allowEditsInFiles`.

**Comment by MrK004** [comment]
I solved this by checking, if the same mods that are already in the normal `mod` folder are also in the Automodpack's `host-modpack` folder.

Ensure that the same mod is **only** either in the `mod` or the `host-modpack` folder.

**Comment by EvasiveUpgrade** [comment]
so i did make sure that the mods folder was not in the alloweditsinfiles 
![image](https://github.com/user-attachments/assets/bee691e4-9557-459f-98b0-d3aedde93828)
and i also verified that the mods were only in the mod folder. it isn't crashing anymore but its also not saving the mods to the mods folder rather to profiles\fabric\automodpack\modpacks\51-222-92-12-51261-automodpack-\mods. i am using modrinth and starting does not move the mods from the download location to the  mods folder to launch with the mods, i manually move the mods and they work but then the crashing starts to happen again

**Comment by EvasiveUpgrade** [comment]
i ended up getting it to work by just making a new profile and having the only mod intitally installed being the automodpack and it ended up working

**Comment by Danszerlied** [comment]
> Solution: dont add `mods` to the editable files on server

Are you planning on fixing that on 1.20.1? I can't find a way to allow edits on some mods if there's a way please tell me

**Comment by Skidamek** [comment]
@Danszerlied i mean you just should not have any mods set as editable by users. 

This issue is still open only to remind me to improve my code related to this isssue to completely prevent this re-restarting. (I could just prevent any mods from being able to be editable but I believe someone could still in some obscure way use this feature)

If you don't have any mods set as editable then you may be experiencing different issue, please describe it in more detail attaching log file. Or just message me on discord, you can find it on the wiki page.

**Comment by Danszerlied** [comment]
> @Danszerlied i mean you just should not have any mods set as editable by users.
> 
> This issue is still open only to remind me to improve my code related to this isssue to completely prevent this re-restarting. (I could just prevent any mods from being able to be editable but I believe someone could still in some obscure way use this feature)
> 
> If you don't have any mods set as editable then you may be experiencing different issue, please describe it in more detail attaching log file. Or just message me on discord, you can find it on the wiki page.

I have some optional client-side mods installed on my server, some members would like to delete it for personal preference reasons.

the issue i'm facing is if I set add a mod to allowEditsInFiles the game goes to the automodpack restart loop upon launch

hope this issue gets fixed soon i could really use this feature

**Comment by Skidamek** [comment]
That's what I meant by "you just should not have any mods set as editable by users"
Editable files are still required files for all players. What's editable does is only allows you to change the file insides but the files itself must still exists.

However I plan implementing feature thanks to which you will be able to create additional optional modpack for e.g. these mods. See #223 and #151

**Comment by Danszerlied** [comment]
> That's what I meant by "you just should not have any mods set as editable by users" Editable files are still required files for all players. What's editable does is only allows you to change the file insides but the files itself must still exists.
> 
> However I plan implementing feature thanks to which you will be able to create additional optional modpack for e.g. these mods. See #223 and #151

Oh okay, thank you for clarifying 👍

**Comment by V7I0SeventhSector** [comment]
> Solution: dont add `mods` to the editable files on the server

so what happens if i have this problem but i have no mods in this? (is this the correct thing?)

"allowEditsInFiles": [
    "/options.txt",
    "/config/*",
    "!/config/excludeThisFile"
  ],

**Comment by Skidamek** [comment]
@V7I0SeventhSector  yes thats correct, please open new issue

**Comment by JustJade2007** [comment]
So wait if I have it set like this does that mean that people cannot use client side mods?

**Comment by Skidamek** [comment]
> So wait if I have it set like this does that mean that people cannot use client side mods?

they always can, no matter the options
